src/app/campaign/flight/flight-zones-form.component.ts

```diff
@@ -129,16 +129,13 @@ export class FlightZonesFormComponent implements ControlValueAccessor, OnDestroy
   }
 
   onAddZone() {
-    // this will send an Add Zone action dispatch up thru the state to be added to the flight,
-    //  and the zone control will be added to the form in writeValue
-    // add the zone that's at the top of the options list
-    this.addZone.emit({ zone: filterZones(this.zoneOptions, this.flightZones as InventoryZone[]).shift() });
```

Comment:
further discussion with @rpeterman-gp determined this is how we're gonna do form arrays, this results in CAMPAIGN_FLIGHT_FORM_UPDATE and CAMPAIGN_FLIGHT_PREVIEW_CREATE because the update will go out through the parent form's valueChanges stream (but only once per update ;)